lib/Conversion/SeqToSV/FirRegLowering.cpp

```diff
@@ -107,53 +115,273 @@ void FirRegLowering::addToIfBlock(OpBuilder &builder, Value cond,
   }
 }
 
+/// Attach an inner-sym to field-id 0 of the given register, or use an existing
+/// inner-sym, if present.
+static StringAttr getInnerSymFor(InnerSymbolNamespace &innerSymNS,
+                                 seq::FirRegOp reg) {
+  auto attr = reg.getInnerSymAttr();
+
+  // If we have an inner sym attribute already, and if there exists a symbol for
+  // field-id 0, then just return that.
+  if (attr)
+    if (auto sym = attr.getSymIfExists(0))
+      return sym;
+
+  // Otherwise, we have to create a new inner sym.
+  auto *context = reg->getContext();
+
+  auto hint = reg.getName();
+
+  // Create our new property for field 0.
+  auto sym = StringAttr::get(context, innerSymNS.newName(hint));
+  auto visibility = StringAttr::get(context, "private");
```

Comment:
FWIW I believe this might be first use of visibility re:inner symbols (first that isn't default of 'public').

Probably we should remove this field until it has meaning.

Looks like some tests make use of it, but I don't think we create any that aren't "public".

Anyway, since it doesn't mean anything presently they aren't /not/ private either so... 👍 .

Reply:
Made it public :+1: